File: server/src/main/java/com/krillsson/sysapi/core/metrics/GpuMetrics.java
package com.krillsson.sysapi.core.metrics;

import com.krillsson.sysapi.core.domain.gpu.Gpu;
import com.krillsson.sysapi.core.domain.gpu.GpuHealth;
import com.krillsson.sysapi.core.domain.gpu.GpuInfo;
import com.krillsson.sysapi.core.domain.gpu.GpuLoad;
import oshi.hardware.Display;

import java.util.List;
import java.util.Map;

public interface GpuMetrics {
    List<Gpu> gpus();
    List<Display> displays();
    List<GpuLoad> gpuLoads();
}

File: src/main/kotlin/com/krillsson/sysapi/core/metrics/GpuMetrics.kt
package com.krillsson.sysapi.core.metrics

import com.krillsson.sysapi.core.domain.gpu.Gpu
import com.krillsson.sysapi.core.domain.gpu.GpuLoad
import oshi.hardware.Display
import reactor.core.publisher.Flux

interface GpuMetrics {
    fun gpus(): List<Gpu>
    fun displays(): List<Display>
    fun gpuLoads(): List<GpuLoad>
    fun gpuLoadById(id: String): GpuLoad?
    fun gpuLoadEvents(): Flux<List<GpuLoad>>
    fun gpuLoadEventsById(id: String): Flux<GpuLoad>
}
